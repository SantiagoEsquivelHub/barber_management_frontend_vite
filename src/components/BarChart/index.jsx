import React from 'react'
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
} from 'chart.js';
import { Line } from 'react-chartjs-2';
import PropTypes from 'prop-types';

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);

/*Objeto con las caracteristicas a mostrar que va a tener el gráfico*/
const options = {
  responsive: true,
  plugins: {
    legend: {
      position: 'top',
    },
    title: {
      display: true,
      text: 'Reporte mensual',
    },
  },
};

/*Componente que muestra una gráfica según la data que llega (se usa en la interna de los barberos y en el Dashboard)*/

const BarChart = ({ data }) => {

  /*Se obtienen los labels y la datas a mostrar, según la data que llega por desestructuración*/
  let labels = data?.map(e => e.nombre_servicio);
  let datas = data?.map(e => e.count);

  let data1 = {
    labels: labels,
    datasets: [
      {
        label: 'Servicios',
        data: datas,
        borderColor: '#fdc43f',
        backgroundColor: '#fdc43f',
      },
    ],
  };

  return (

    <Line options={options} data={data1} />

  )
}

BarChart.propTypes = {
  data: PropTypes.array.isRequired
}

export default BarChart;