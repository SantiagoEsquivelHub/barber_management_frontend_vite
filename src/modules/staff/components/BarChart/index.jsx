import React from 'react'
import {
    Chart as ChartJS,
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend,
  } from 'chart.js';
  import { Line } from 'react-chartjs-2';
  import faker from 'faker';
  
  ChartJS.register(
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend
  );
  
  export const options = {
    responsive: true,
    plugins: {
      legend: {
        position: 'top',
      },
      title: {
        display: true,
        text: 'Servicios brindados este mes',
      },
    },
  };
  
  const labels = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '27', '28', '29', '30'];
  


const BarChart = ({ data }) => {
const labels = data.map(e=> e.nombre_servicio);
const datas = data.map(e=> e.count);

  let data1 = {
    labels: labels,
    datasets: [
      {
        label: 'Servicios',
        data: datas,
        borderColor: '#fdc43f',
        backgroundColor: '#fdc43f',
      },
    ],
  };

    return (
      
            <Line options={options} data={data1} />
        
    )
}

export default BarChart;