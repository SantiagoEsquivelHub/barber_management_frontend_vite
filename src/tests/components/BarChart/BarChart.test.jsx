import { render } from '@testing-library/react'
import BarChart from '../../../components/BarChart';
import useResizeObserver from "use-resize-observer";

window.ResizeObserver =
    window.ResizeObserver ||
    jest.fn().mockImplementation(() => ({
        disconnect: jest.fn(),
        observe: jest.fn(),
        unobserve: jest.fn(),
    }));

describe('Pruebas en <BarChart />', () => {

    test('Debe hacer match con el Snapshot', () => {

        const data = [
            { nombre_servicio: 'Corte de cabello', count: 1 },
            { nombre_servicio: 'Barba', count: 7 },
            { nombre_servicio: 'Corte de cabello y Barba', count: 3 },

        ]
        const { container } = render(<BarChart data={data} />);
        expect(container).toMatchSnapshot();
    })

})