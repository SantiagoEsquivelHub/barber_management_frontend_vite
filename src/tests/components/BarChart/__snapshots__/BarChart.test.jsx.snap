// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <BarChart /> Debe hacer match con el Snapshot 1`] = `
<div>
  <canvas
    height="0"
    role="img"
    style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
    width="0"
  />
</div>
`;
